targets/mbedos5/jerryscript-mbed/jerryscript-mbed-util/wrappers.h

```diff
@@ -0,0 +1,93 @@
+/* Copyright (c) 2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef _JERRYSCRIPT_MBED_UTIL_WRAPPERS_H
+#define _JERRYSCRIPT_MBED_UTIL_WRAPPERS_H
+
+/*
+ * Used in header/source files for wrappers, to declare the signature of the
+ * registration function.
+ */
+#define DECLARE_JS_WRAPPER_REGISTRATION(NAME) \
+  void jsmbed_wrap_registry_entry__ ## NAME (void)
+
+//
+// 2. Wrapper function declaration/use macros
+//
+
+// Global functions
+#define DECLARE_GLOBAL_FUNCTION(NAME) \
+jerry_value_t \
+NAME_FOR_GLOBAL_FUNCTION(NAME) (const jerry_value_t function_obj_p, \
+                  const jerry_value_t  this_obj, \
+                  const jerry_value_t    args[], \
+                  const jerry_length_t   args_count)
+
+#define REGISTER_GLOBAL_FUNCTION(NAME) \
+  jsmbed_wrap_register_global_function ( # NAME, NAME_FOR_GLOBAL_FUNCTION(NAME) )
+
+// Class constructors
+#define DECLARE_CLASS_CONSTRUCTOR(CLASS) \
+jerry_value_t \
+NAME_FOR_CLASS_CONSTRUCTOR(CLASS) (const jerry_value_t function_obj, \
+                  const jerry_value_t    this_obj, \
+                  const jerry_value_t    args[], \
+                  const jerry_length_t   args_count)
+
+#define REGISTER_CLASS_CONSTRUCTOR(CLASS) \
+  jsmbed_wrap_register_class_constructor ( # CLASS, NAME_FOR_CLASS_CONSTRUCTOR(CLASS) )
+
+// Class functions
+#define DECLARE_CLASS_FUNCTION(CLASS, NAME) \
+jerry_value_t \
+NAME_FOR_CLASS_FUNCTION(CLASS, NAME) (const jerry_value_t function_obj, \
+                  const jerry_value_t  this_obj, \
+                  const jerry_value_t    args[], \
+                  const jerry_length_t   args_count)
+
+#define ATTACH_CLASS_FUNCTION(OBJECT, CLASS, NAME) \
+  jsmbed_wrap_register_class_function (OBJECT, # NAME, NAME_FOR_CLASS_FUNCTION(CLASS, NAME) )
+
+//
+// 3. Argument checking macros
+//
+#define CHECK_ARGUMENT_COUNT(CLASS, NAME, EXPR) \
+  if (!(EXPR)) { \
+    LOG_PRINT_ALWAYS("ERROR: wrong argument count for %s.%s, expected %s.\n", # CLASS, # NAME, # EXPR ); \
+    return false; \
```

Comment:
You're correct - this code is a hangover from an older version of the jerryscript API, and GCC is casting the false value to jerry_value_t, which is definitely bad. I'll fix this, and similar macros too.